### PR TITLE
Checkbox - render tests

### PR DESCRIPTION
**Pull request by M-i-k-e-l**
## Description
Checkbox - render tests

## Changelog
Checkbox - render tests

### CODE REVIEW

File: src/components/checkbox/__tests__/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import React, {useState, useCallback} from 'react';
+import {render, fireEvent} from '@testing-library/react-native';
+import Checkbox from '../index';
+
+export const testID = 'checkbox';
```
Comment (Inbal-Tish) [inline src/components/checkbox/__tests__/index.spec.js:5]: Why are you exporting the testID?

File: src/components/checkbox/__tests__/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+  return <Checkbox {...others} onValueChange={_onValueChange} value={_value} testID={testID}/>;
+};
+
+describe('Checkbox renderer test', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('Default value is false', async () => {
```
Comment (Inbal-Tish) [inline src/components/checkbox/__tests__/index.spec.js:26]: Why your tests are async?

File: src/components/checkbox/__tests__/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    const {getByTestId} = render(<TestCase {...props}/>);
+
+    const checkbox = getByTestId(testID);
+    fireEvent.press(checkbox);
+    expect(onValueChange).toHaveBeenCalledTimes(1);
+    expect(onValueChange).toHaveBeenCalledWith(true);
+  });
+
+  it('Send value (false)', async () => {
```
Comment (Inbal-Tish) [inline src/components/checkbox/__tests__/index.spec.js:37]: Why you need this test? You checked the default (false) in the previous test, are you checking react's passed prop mechanism or there's a different behavior when passing false over a default false?

File: src/components/checkbox/__tests__/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    const {getByTestId} = render(<TestCase {...props}/>);
+
+    const checkbox = getByTestId(testID);
+    fireEvent.press(checkbox);
+    expect(onValueChange).not.toHaveBeenCalled();
+  });
+
+  it('Disabled (value = true)', async () => {
+    const props = {onValueChange, value: true, disabled: true};
+    const element = <TestCase {...props}/>;
```
Comment (Inbal-Tish) [inline src/components/checkbox/__tests__/index.spec.js:86]: You can pass on the 'element' const and extract getByTestId directly (like you did in the other tests)

File: src/components/checkbox/__tests__/index.spec.js (hunk 1 above)
Comment (M-i-k-e-l) [reply src/components/checkbox/__tests__/index.spec.js:5]: I think I've put the `TestCase` in a different file and it was copy-pasted, 10x

File: src/components/checkbox/__tests__/index.spec.js (hunk 2 above)
Comment (M-i-k-e-l) [reply src/components/checkbox/__tests__/index.spec.js:26]: I blame copy-pasted again. 10x

File: src/components/checkbox/__tests__/index.spec.js (hunk 3 above)
Comment (M-i-k-e-l) [reply src/components/checkbox/__tests__/index.spec.js:37]: I think I prefer to test explicitly as well as default behaviour, not sure what the best practice, do you want to ask the team?

File: src/components/checkbox/__tests__/index.spec.js (hunk 4 above)
Comment (M-i-k-e-l) [reply src/components/checkbox/__tests__/index.spec.js:86]: Missed that, 10x

File: src/components/checkbox/__tests__/index.spec.js (hunk 3 above)
Comment (Inbal-Tish) [reply src/components/checkbox/__tests__/index.spec.js]: Sure, we can ask them. Anyways Let's leave it in the meantime

**Review by Inbal-Tish — APPROVED** [review]

**Pull request merged**